thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java

```diff
@@ -145,6 +154,7 @@ public YamlResource() {
     this.evaluationDAO = DAORegistry.getInstance().getEvaluationManager();
     this.sessionDAO = DAORegistry.getInstance().getSessionDAO();
     this.yaml = new Yaml();
+    this.executor = Executors.newFixedThreadPool(PREVIEW_PARALLELISM);
```

Comment:
How about a using a `newCachedThreadPool`? That way the calls won't easily timeout at peak traffic.

Reply:
Actually, in this case, we want to use a `newFixedThreadPool`, because we would like to control the maximum number of preview tasks to keep it from eating all dashboard resource.